src/jit/flowgraph.cpp

```diff
@@ -21486,6 +21496,92 @@ void                Compiler::fgInline()
 #endif // DEBUG
 }
 
+#ifdef DEBUG
+
+//------------------------------------------------------------------------
+// fgNotInlineCandidate: walk tree to scan for calls that were left
+// embedded because they weren't inline candidates.
+//
+// Arguments:
+//    pTree - pointer to pointer tree node being walked
+//    data  - contextual data for the walk
+//
+// Return Value:
+//    walk result
+//
+// Note:
+//    Only used in debug to scan trees for calls embedded in trees
+//    that were not classified as candidates.
+
+Compiler::fgWalkResult      Compiler::fgFindNotInlineCandidate(GenTreePtr* pTree,
+                                                               fgWalkData* data)
+{
+    GenTreePtr tree = *pTree;
+    if (tree->gtOper == GT_CALL)
+    {
+        Compiler*    compiler = data->compiler;
+        GenTreePtr   stmt     = (GenTreePtr) data->pCallbackData;
+        GenTreeCall* call     = tree->AsCall();
+
+        compiler->fgNotInlineCandidate(stmt, call);
+    }
+    return WALK_CONTINUE;
+}
+
+//------------------------------------------------------------------------
+// fgNotInlineCandidate: account for inlining failures in calls
+// not marked as inline candidates.
+//
+// Arguments:
+//    tree  - statement containing the call
+//    call  - the call itself
+//
+// Notes:
+//    Used in debug only to try and place descriptions of inline failures
+//    into the proper context in the inline tree.
+
+void Compiler::fgNotInlineCandidate(GenTreePtr   tree,
+                                    GenTreeCall* call)
+{
```

Comment:
Would it make sense for this to be "fgSetNotInlineCandidate"? or "fgMarkNotInlineCandidaite"? At first from the name I thought it would return a Boolean result, then I realized it is marking it.